toolkit/partner-chains-cli/src/automatic_generate_keys/mod.rs

```diff
@@ -0,0 +1,441 @@
+
+#[derive(Clone, Debug, clap::Parser)]
+pub struct AutomaticGenerateKeysCmd {
+    /// Substrate node RPC URL
+    #[arg(long, default_value = "http://localhost:9944")]
+    pub url: String,
+}
+
+#[derive(Debug)]
+pub struct AutomaticGenerateKeysConfig {
+    pub node_url: String,
+}
+
+impl AutomaticGenerateKeysConfig {
+    pub(crate) fn load<C: IOContext>(_context: &C, url: String) -> Self {
+        Self {
+            node_url: url,
+        }
+    }
+}
+
+// Generic session keys type - list of (key_type_id, public_key) pairs
+pub type SessionKeys = Vec<KeyPair>;
+pub type KeyPair = ([u8; 4], Vec<u8>);
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SessionKeyInfo {
+    pub key_type: String,
+    pub public_key: String,
+}
+
+#[derive(Debug)]
+struct SessionKeyTypeInfo {
+    key_type_id: [u8; 4],
+    key_name: String,
+    key_size: usize,
+}
+
+impl CmdRun for AutomaticGenerateKeysCmd {
+    fn run<C: IOContext>(&self, context: &C) -> anyhow::Result<()> {
+        context.print("🔐 Generating session keys automatically using runtime metadata...");
+        context.enewline();
+
+        let config = AutomaticGenerateKeysConfig::load(context, self.url.clone());
+        context.eprint(&format!("🔗 Connecting to node at: {}", config.node_url));
+
+        let keys = generate_keys_via_runtime(&config, context)?;
+        context.enewline();
+
+        // Convert to output format and print as JSON
+        let session_keys_info: Vec<SessionKeyInfo> = keys
+            .iter()
+            .map(|(key_type_id, public_key)| {
+                let key_type_str = String::from_utf8_lossy(key_type_id).to_string();
+                SessionKeyInfo {
+                    key_type: key_type_str.clone(),
+                    public_key: hex::encode(public_key),
+                }
+            })
+            .collect();
+
+        // Output as JSON
+        let json_output = serde_json::to_string_pretty(&session_keys_info)
+            .context("Failed to serialize session keys to JSON")?;
+
+        context.print(&json_output);
+        context.enewline();
+
+        context.eprint("🚀 All done!");
+
+        Ok(())
+    }
+}
+
+fn generate_keys_via_runtime<C: IOContext>(
+    config: &AutomaticGenerateKeysConfig,
+    context: &C,
+) -> Result<SessionKeys> {
+    context.eprint("⚙️ Connecting to node and analyzing runtime...");
+    
+    // Create a new Tokio runtime for the async operation
+    let rt = tokio::runtime::Runtime::new().context("Failed to create Tokio runtime")?;
+    
+    let keys = rt.block_on(async {
+        // Connect to the node using subxt to get metadata
+        let client = OnlineClient::<PolkadotConfig>::from_url(&config.node_url)
+            .await
+            .context("Failed to connect to node")?;
+        
+        context.eprint("✅ Connected to node successfully");
+        context.eprint("🔍 Extracting SessionKeys type definition from runtime metadata...");
```

Comment:
this step is not needed anymore. We can get ready list by invoking RPC command on the node